### PR TITLE
defensive coding fix for set_face_set_gcs_list_from_dict function

### PR DESCRIPTION
**Pull request by andy-beer**
resolves issue #401

### CODE REVIEW

File: resqpy/grid/face_functions.py [hunk 1]
```diff
@@ -93,8 +93,15 @@ def set_face_set_gcs_list_from_dict(grid, face_set_dict = None, create_organizin
...
-    for feature in face_set_dict:
+    for feature, kelp_values in face_set_dict.items():
         gcs = rqf.GridConnectionSet(grid.model, grid = grid)
+        if len(kelp_values) == 2:
+            kelp_j, kelp_i = kelp_values
+            axis = 'K'
+        elif len(kelp_values) == 3:
+            kelp_j, kelp_i, axis = face_set_dict[feature]
+        else:
+            raise Exception('grid face set dictionary item messed up')
```
Comment (maxm17) [inline resqpy/grid/face_functions.py:104]: I would change the Exception to a ValueError

**Review by maxm17 — CHANGES_REQUESTED** [review]

**Comment by andy-beer** [comment]
Thanks for those comments @maxm17 – all 3 changes made as suggested.

**Comment by andy-beer** [comment]
Max, I couldn't find the unit test that had flagged up this bug.  Maybe I was looking in the wrong branch.  Anyway, could you please track down the test and run it against this branch?  Or if that is not convenient, I think we can safely merge this branch into master.

**Comment by maxm17** [comment]
@andy-beer It may be better if you could check that it is working correctly, as I don't know what is expected. The test is in master: unit_tests/grid/test_faults.py::test_find_faults_create_organizing_objects_where_needed_true. You will need to uncomment the act and assertion lines

**Review by maxm17 — APPROVED** [review]

**Pull request merged**